---
layout: page
title: About
permalink: /about/
---

# copermut

[permut.co](http://permut.co) – Web project on information theory, graph patterns, permutations, … copermutations

> `(478)(356) ⪯ `:computer:

### Aims:
- __self-hosted__ as a first project on my own server
- __built with Structr__ parallel [Neo4j](https://neo4j.com) database (else [Cayley](https://github.com/cayleygraph/cayley))
- __design and form graph database of concepts__ named entities as well as Wiki-structured map

### Long term goals:

I'd like to 'manage' the abstract/'deep' connections inferred through a [hypergraph](https://en.wikipedia.org/wiki/Hypergraph) database (e.g. _Structr_) 'on top of' a graph database (mapping items to one another, with some level of reasoning and accountability in doing so). In a roundabout way, this is what classic web companies such as Facebook, Twitter and Google do on my behalf already.

As much as I try to make the most of these services for research, the models they build of my web experience is locked away from view, so I'll be exploring other ways of getting insight on knowledge, as it exists 'in the real world' of my interactions with thoughts and ideas.

The project outlined above could be quite powerful as a deliberately implemented personal research tool (which isn't really something I've seen attempted).

- to study your own learning/attention processes ('explanatory' modelling), and/or those of readers of some output from them (use your imagination really)
  - I've made bots for research literature updates to Twitter in the past, and you can now get 'feedback' on the attention given to their output on a per-post basis, which could - maybe - guide a development, if not quite machine learning, process. Two concepts along this line of thought are: [_queck_](https://github.com/lmmx/queck), and [_nat_](https://github.com/lmmx/nat/wiki) (more details TBC)

  - machine learning models (I use _TensorFlow_ mainly) could extend the 'understanding' from simple links to some level of reason. In the long run, I'd like for some means for a system like this to anticipate things I'd like to read/know about, and to try using computational logic systems


### Research-assistive application:
- Biological research-oriented spin-off with the same format, use of biological linked data ("[ontologies](http://obofoundry.org)")

- see: [Wiki](https://github.com/lmmx/copermut/wiki) consolidating readings and discussions of {category,graph,order} theory, {geometric,linear,matrix} algebra, {ML,DL,NLP} — _etc._ — as imaginative routes and progressive community resource in systems biology, genomic statistics, and their surrounding communities in the computer, life, and physical sciences.

# About the author

Louis Maddox, BSc Biochemistry 2015, MSc Bioinformatics and Systems Biology 2015-16

I graduated with a degree in Biochemistry in 2015 after an undergraduate stint at the University of Manchester, and a summer’s research placement with the Babu group at the MRC-LMB in Cambridge, which came about thanks to shared interests in intrinsically disordered proteins, the topic of my 2nd year dissertation, and various posts on my blog, [_biochemistri.es_](http://biochemistri.es).

During undergraduate study I fell down the rabbit hole of programming, and in my final year discovered the dynamical systems biology of ultradian [4-8 hour periodic] oscillations in gene expression. Cyclic gene regulation, I learnt, could be expressed through geometrical language, and had broad crossovers to physics and other fields. Within this I hunted out candidate miRNAs from microarray datasets, and stayed on to continue readings in this field for a Masters degree.

I’m currently working on research projects relating to cancer genetics within the University of Manchester Masters in Bioinformatics and Systems Biology, and am seeking a PhD in mathematical/computational systems biology, as far as my abilities can take me really.

Towards this end, this site is an attempt to consolidate the broad readings (almost chaotically so at times) which encompass:

- __biochemical engineering__: beyond the fundamental biochemical and life sciences, systems/network approaches, and information theory [the 'informatics' in 'bioinformatics']) there are untapped areas of statistical physics, such as interfaces and colloids, which are forming a route to biophysics and nanoscience
- __engineering mathematics__
- __combinatorics__ I was mesmerised by _permutation_ through the writings of Greg Egan around the time of graduation, and later pulled by force into the application of permutation graphs in reading a biography of Pascal this year.
- __informatics__
- __machine learning and deep learning__ ("the only game in town") as an academic joked to me last month. GPUs and CUDA
  - __computer vision__
- __graphics__: GPUs once again: I'd currently like to learn more about _Vulkan API_, the successor to _OpenGL_
- __graphs__ (mathematical formulation, databases, category theory, algorithms on graphs)

An overarching theme is of 'usufructuary spaces' for research, and 'copermutation' (a long-winded, medieval French way of saying 'public good' as detailed in the notes herein). For obvious reasons this chimes with my goals as a scientist, but moreover I feel at this point that I need to actively construct a mode of knowledge work that will weave these complex threads into a more consistent picture.

Mentorship is greatly valued, so I'm putting these notes out here to maximise potential conversation as well as utility.

Find me on Twitter @biochemistries, as well as @permutans (my separate account for tech/programming-related goings on). @permutans is also where you can find me on the Telegram messaging service if you'd like to chat about the research discussed, request help, or propose a collaboration.
